nomad/rpc.go

```diff
@@ -98,22 +99,44 @@ func (r *rpcHandler) listen(ctx context.Context) {
 			if r.shutdown {
 				return
 			}
-
-			select {
-			case <-ctx.Done():
-				return
-			default:
-			}
-
-			r.logger.Error("failed to accept RPC conn", "error", err)
+			r.handleAcceptErr(err, ctx)
 			continue
 		}
+		// No error, reset loop delay
+		r.acceptLoopDelay = 0
 
 		go r.handleConn(ctx, conn, &RPCContext{Conn: conn})
 		metrics.IncrCounter([]string{"nomad", "rpc", "accept_conn"}, 1)
 	}
 }
 
+// Sleep to avoid spamming the log, with a maximum delay according to whether or not the error is temporary
+func (r *rpcHandler) handleAcceptErr(err error, ctx context.Context) {
+	const baseAcceptLoopDelay = 5 * time.Millisecond
+	const maxAcceptLoopDelay = 5 * time.Second
+	const maxAcceptLoopDelayTemporaryError = 1 * time.Second
```

Comment:
You can shorten the names since they're locals: baseDelay, maxDelay, maxDelayTempError

No need to change.